java/custom/com/nokia/mid/ui/TextEditor.java

```diff
@@ -3,29 +3,20 @@
 import javax.microedition.lcdui.Font;
 import java.util.Hashtable;
 
-class TextEditorThread implements Runnable {
+class TextEditorThread extends Thread {
     // We need a thread to be able to wake from js when there is an async js keyboard event.
-    native void sleep();
-    native int getNextDirtyEditor();
-
-    Hashtable listeners;
+    native TextEditor getNextDirtyEditor();
 
     TextEditorThread() {
-        listeners = new Hashtable();
+        setPriority(Thread.MAX_PRIORITY);
     }
 
     public void run() {
         while (true) {
-            sleep();
-            int dirty = getNextDirtyEditor();
-            TextEditor t = (TextEditor)listeners.get("" + dirty);
-            t.myListener.inputAction(t, TextEditorListener.ACTION_CONTENT_CHANGE);
+            TextEditor dirty = getNextDirtyEditor();
+            dirty.myListener.inputAction(dirty, TextEditorListener.ACTION_CONTENT_CHANGE);
```

Comment:
Theoretically dirty.myListener could be null, though all the midlets we tests  so far have non null myListener.
I suggest add a null check to take precaution.